TradeItIosEmsApi/TradeItRequestResultFactory.m

```diff
@@ -69,6 +74,25 @@ + (NSString *)getHostForEnvironment:(TradeitEmsEnvironments)env {
 + (NSMutableURLRequest *)buildJsonRequestForModel:(JSONModel *)requestObject
                                         emsAction:(NSString *)emsAction
                                       environment:(TradeitEmsEnvironments)env {
+    if (self.requestFactory != nil) {
+        NSURL *url = [NSURL URLWithString:emsAction
+                            relativeToURL:[TradeItRequestResultFactory getBaseUrlForEnvironment:env]];
+
+        NSDictionary *requestParams = [requestObject toDictionary];
+
+        NSDictionary *headers = @{
+                                  @"Accept": @"application/json",
+                                  @"Content-Type": @"application/json"
+                                  };
+
+        NSURLRequest *request = [TradeItRequestResultFactory.requestFactory buildRequestForUrl:url
+                                                                                    httpMethod:@"POST"
+                                                                                    parameters:requestParams
+                                                                                       headers:headers];
+
+        return (NSMutableURLRequest *)[request mutableCopy];
+    }
```

Comment:
I still feel like we should have a default request factory. This isn't going to go away any time soon so it's useful for us to have single code paths that are more likely to be tested before we push out builds. If something broke unexpectedly here, we wouldn't see it until someone creating a `requestFactory` told us. Also is there anything we can/need to do to future proof this when if we rewrite this in Swift?